Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs

```diff
@@ -228,14 +229,32 @@ private List<(EntityUid Entity, CargoPalletComponent Component)> GetCargoPallets
 
     #region Station
 
-    private void SellPallets(EntityUid gridUid, out double amount)
+    private void SellPallets(EntityUid gridUid, EntityUid? station, out double amount)
     {
+        station ??= _station.GetOwningStation(gridUid);
         GetPalletGoods(gridUid, out var toSell, out amount);
 
         _sawmill.Debug($"Cargo sold {toSell.Count} entities for {amount}");
 
         foreach (var ent in toSell)
         {
+            if (station != null)
+            {
+                var ev = new EntitySoldEvent(station.Value);
+                RaiseLocalEvent(ent, ref ev);
+
+                if (TryComp<ContainerManagerComponent>(ent, out var containers))
+                {
+                    foreach (var container in containers.Containers.Values)
+                    {
+                        foreach (var containedEntity in container.ContainedEntities)
+                        {
+                            RaiseLocalEvent(containedEntity, ref ev);
+                        }
+                    }
+                }
+            }
+
```

Comment:
Can't you just raise 1 event with all the tosell entities and iterate it yourself? Gonna save a lot of events going out.